packages/cli/src/utils/foundry.ts

```diff
@@ -1,24 +1,99 @@
-    const { stdout } = await execa(command, args, { stdout: "pipe", stderr: "pipe" });
-    return stdout;
-  } catch (error: any) {
-    if (error?.stderr) {
-      console.error(error.stderr);
-    }
-    console.error(chalk.red(`Error running "${command} ${args.join(" ")}"`));
-    process.exit(1);
-  }
+export interface ForgeConfig {
+  // project
+  src: string;
+  test: string;
+  script: string;
+  out: string;
+  libs: string[];
+  eth_rpc_url: string | null;
+
+  // all unspecified keys (this interface is far from comprehensive)
+  [key: string]: unknown;
+}
+
+/**
+ * Get forge config as a parsed json object.
+ */
+export async function getForgeConfig(profile?: string) {
+  const { stdout } = await execa("forge", ["config", "--json"], {
+    stdio: ["inherit", "pipe", "pipe"],
+    env: { FOUNDRY_PROFILE: profile },
+  });
+
+  return JSON.parse(stdout) as ForgeConfig;
+}
+
+/**
+ * Get the value of "src" from forge config.
+ * The path to the contract sources relative to the root of the project.
+ */
+export async function getSrcDirectory(profile?: string) {
+  return (await getForgeConfig(profile)).src;
+}
+
+/**
+ * Get the value of "script" from forge config.
+ * The path to the contract sources relative to the root of the project.
+ */
+export async function getScriptDirectory(profile?: string) {
+  return (await getForgeConfig(profile)).script;
+}
+
+/**
+ * Get the value of "test" from forge config.
+ * The path to the test contract sources relative to the root of the project.
+ */
+export async function getTestDirectory(profile?: string) {
+  return (await getForgeConfig(profile)).test;
+}
+
+/**
+ * Get the value of "out" from forge config.
+ * The path to put contract artifacts in, relative to the root of the project.
+ */
+export async function getOutDirectory(profile?: string) {
+  return (await getForgeConfig(profile)).out;
+}
+
+/**
+ * Get the value of "eth_rpc_url" from forge config, default to "http://127.0.0.1:8545"
+ * @param profile The foundry profile to use
+ * @returns The rpc url
+ */
+export async function getRpcUrl(profile?: string) {
+  return (await getForgeConfig(profile)).eth_rpc_url || "http://127.0.0.1:8545";
 }
 
-export async function forge(...args: string[]): Promise<string> {
-  return safeExec("forge", args);
+/**
+ * Execute a forge command
+ * @param args The arguments to pass to forge
+ * @param options { profile?: The foundry profile to use; silent?: If true, nothing will be logged to the console }
+ * @returns Stdout of the command
+ */
+export async function forge(args: string[], options?: { profile?: string; silent?: boolean }): Promise<string> {
+  return options?.silent
+    ? (
+        await execa("forge", args, {
+          env: { FOUNDRY_PROFILE: options?.profile },
+          stdout: "pipe",
+          stderr: "pipe",
+        })
+      ).stdout
+    : execLog("forge", args, {
+        env: { FOUNDRY_PROFILE: options?.profile },
```

Comment:
all environment variables are part of `process.env` (because we use `dotenv` in `mud.ts`), and everything that is part of `process.env` is automatically included in `execa`'s `env`. This means we actually wouldn't need a `--profile` flag in our cli, but I think it's still useful to explicitly know the profile in our cli, so we can 1. log it to the user and 2. add more advanced features later, like auto-funding accounts via faucets if a certain profile is used (which could arguably also be done via a check on the rpc url, but I still think being able to set the profile via a cli param in addition to falling back to the env variable is good ux).

Reply:
Personally I find the ux of using a flag a bit nicer than writing `FOUNDRY_PROFILE=profile` in front of the comment, but this is not a strong opinion. The other thing I was envisioning was adding default options based on the profile arg itself, instead of requiring them in the foundry config. (Eg `--profile lattice-testnet` -> default values for rpc, auto-funding the deployer wallet from the faucet, etc). Using the `foundry.toml` / `FOUNDRY_PROFILE` env variable feels like an unnecessary indirection here. We could separate that use-case from the "foundry profile" arg though. In any case I propose doing this in a follow up.

Reply:
Oh one other thing: these are utils, where a consumer might want to explicitly execute one of them with a profile other than the one found in `FOUNDRY_PROFILE`. Defaulting to the `FOUNDRY_PROFILE` happens one level higher in the `command` itself.

Reply:
I'm good with all that! Foundry basically has the same: env vars + matching command args.